primitives/arithmetic/src/helpers_128bit.rs

```diff
@@ -182,6 +183,19 @@ mod double128 {
 	}
 }
 
+pub const fn checked_multiply_by_rational_with_rounding(
+	a: u128,
+	b: u128,
+	c: u128,
+	r: Rounding,
+) -> Result<Option<u128>, &'static str> {
+	if c == 0 {
+		return Err("Division by zero")
+	}
+
+	Ok(multiply_by_rational_with_rounding(a, b, c, r))
```

Comment:
As a general consideration of design:  
I think its easier to put the actual logic inside the `checked_` function such that the unchecked variant is just `checked_function().expect()` or `checked_function.unwrap_or(whatever)` since otherwise you have the case like now, where the checked variant does not have any logic and can therefore not ensure that it wont ever panic.